sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEvent.kt

```diff
@@ -0,0 +1,158 @@
+package io.sentry.android.okhttp
+
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SpanStatus
+import io.sentry.TypeCheckHint
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.CONNECTION_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.CONNECT_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.REQUEST_BODY_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.REQUEST_HEADERS_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.RESPONSE_BODY_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.RESPONSE_HEADERS_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.SECURE_CONNECT_EVENT
+import io.sentry.util.UrlUtils
+import okhttp3.Call
+import okhttp3.Response
+import java.net.URL
+
+private val uuidRegex by lazy { Regex("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}") }
+
+internal class SentryOkHttpEvent(private val hub: IHub, private val call: Call) {
+    private val eventSpans: MutableMap<String, ISpan> = HashMap()
+    private val breadcrumb: Breadcrumb
+    internal val callRootSpan: ISpan?
+    private var response: Response? = null
+
+    init {
+        val urlDetails = UrlUtils.parse(call.request().url.toString())
+        val url = urlDetails.urlOrFallback
+        val trimmedUrl: String = trimUrl(url)
+        val host: String = call.request().url.host
+        val encodedPath: String = call.request().url.encodedPath
+        val method: String = call.request().method
+
+        // We start the call span that will contain all the others
+        callRootSpan = hub.span?.startChild("http.client", "$method $url")
+
+        urlDetails.applyToSpan(callRootSpan)
+
+        // We setup a breadcrumb with all meaningful data
+        breadcrumb = Breadcrumb.http(url, method)
+        breadcrumb.setData("url", url)
+        breadcrumb.setData("filtered_url", trimmedUrl)
```

Comment:
should we keep this `filtered_url` or should we just put this value in `url`?

Reply:
the difference is basically removing any uuid from the url. 
I copied it from an old pr from Manoel, and thought it could make sense to add it here, but now I'm not so sure. No one complained, so i guess there's no concern of PII here, right?
Guess I can remove it in this case

Reply:
I'm fine with keeping it as well, but I'd then just put it into `url` - otherwise I don't see how we're preventing from leaking PII if we're still reporting the plain url alongside the filtered one :) Also, instead of just removing the uuid, I'd replace it with `[Filtered]` as we do for other stuff.

But I'm leaning towards just keeping it as is and not stripping the uuid

Reply:
Same as https://github.com/getsentry/sentry-java/pull/2659#discussion_r1175044364
We don't need `filtered_url` since the `url` will be split out in different metadata and Relay will strip out what is PII, check RFC and Dart PR as an example.